bus/explorer.go

```diff
@@ -0,0 +1,81 @@
+package bus
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"go.sia.tech/renterd/api"
+)
+
+type (
+	// An explorer retrieves data about the Sia network from an external source.
+	Explorer interface {
+		Enabled() bool
+		BaseURL() string
+		SiacoinExchangeRate(ctx context.Context, currency string) (rate float64, err error)
+	}
+
+	explorer struct {
+		enabled bool
+		url     string
+	}
+)
+
+// NewExplorer returns a new Explorer.
+func NewExplorer(url string, enabled bool) Explorer {
+	return &explorer{
+		enabled: enabled,
+		url:     url,
+	}
+}
+
+// BaseURL returns the base URL of the Explorer.
+func (e *explorer) BaseURL() string {
```

Comment:
I don't think this is used?

Reply:
Stupid merge oversight, thanks. It's exposed in the `/state` route, again to be consistent with `hostd`.